library-best-practices/JLBP-5.md

```diff
@@ -36,8 +36,14 @@ Example 2: There are multiple artifacts that provide classes under
     might be used anywhere that is not a servlet. Define these status codes
     yourself or choose a different library to provide them.
 
-Problems caused by overlapping classes can be extremely difficult to resolve.
-If at all possible, eliminate all but one of the
-overlapping dependencies. Otherwise pay very close attention to which version of
-each overlapping class is chosen. Make sure the project does not depend on any
-behavior or API of the class that is not selected.
+In Java 9 and later overlapping classes become compile-time and runtime errors when
+named modules are used. It is critical, especially in Java 9 and later,
+to remove all but one of the overlapping classes from the classpath.
+
+If possible, upgrade all dependencies on the overlapping classes
+to import them only by the same group ID and artifact ID.
+
+If this isn't possible, for instance because a dependency that imports the
+renamed artifact is unmaintained, then add
+[dependency exclusions](https://maven.apache.org/guides/introduction/introduction-to-optional-and-excludes-dependencies.html) for all but one of the renamed artifacts
```

Comment:
I would suggest "undesired" instead of "renamed"

(It's entirely possible that a "renamed" artifact becomes the desired one)

Reply:
done